pkg/scheduler/core/generic_scheduler.go

```diff
@@ -462,8 +461,8 @@ func (g *genericScheduler) findNodesThatFit(pod *v1.Pod, nodes []*v1.Node) ([]*v
 	if len(g.predicates) == 0 {
 		filtered = nodes
 	} else {
-		allNodes := g.cache.NodeTree().AllNodes()
-		numNodesToFind := g.numFeasibleNodesToFind(int32(len(allNodes)))
+		allNodes := int32(g.cache.NodeTree().NumNodes())
```

Comment:
I assume holding lock over the course of checkNode execution would defeat the purpose of the original improvement.